client/my-sites/stats/controller.js

```diff
@@ -580,41 +584,26 @@ module.exports = {
 			period = rangeOfPeriod( activeFilter.period, date );
 			endDate = period.endOf.format( 'YYYY-MM-DD' );
 
-			switch ( context.params.module ) {
-
+			const module = context.params.module;
+			let fetchActions;
+			switch ( module ) {
 				case 'posts':
-					visitsList = new StatsList( { statType: 'statsVisits', unit: activeFilter.period, siteID: siteId, quantity: 10, date: endDate } );
-					summaryList = new StatsList( { statType: 'statsTopPosts', siteID: siteId, period: activeFilter.period, date: endDate, max: 0, domain: site.slug } );
-					break;
-
-				case 'referrers':
-					summaryList = new StatsList( { siteID: siteId, statType: 'statsReferrers', period: activeFilter.period, date: endDate, max: 0, domain: site.slug } );
-					break;
-
-				case 'clicks':
-					summaryList = new StatsList( { statType: 'statsClicks', siteID: siteId, period: activeFilter.period, date: endDate, max: 0, domain: site.slug } );
-					break;
-
-				case 'countryviews':
-					summaryList = new StatsList( { siteID: siteId, statType: 'statsCountryViews', period: activeFilter.period, date: endDate, max: 0, domain: site.slug } );
-					break;
-
-				case 'authors':
-					summaryList = new StatsList( { statType: 'statsTopAuthors', siteID: siteId, period: activeFilter.period, date: endDate, max: 0, domain: site.slug } );
-					break;
-
-				case 'videoplays':
-					summaryList = new StatsList( { statType: 'statsVideoPlays', siteID: siteId, period: activeFilter.period, date: endDate, max: 0, domain: site.slug } );
+					fetchActions = [
+						{ statType: 'statsVisits', options: {
+							unit: activeFilter.period, quantity: 10, date: endDate
+						} },
+						{ statType: 'statsTopPosts', options: {
+							period: activeFilter.period,
+							date: endDate,
+							max: 0
+						} }
+					];
 					break;
-
 				case 'videodetails':
-					summaryList = new StatsList( { statType: 'statsVideo', post: queryOptions.post, siteID: siteId, period: activeFilter.period, date: endDate, max: 0, domain: site.slug } );
-					break;
-
-				case 'searchterms':
-					summaryList = new StatsList( { siteID: siteId, statType: 'statsSearchTerms', period: activeFilter.period, date: endDate, max: 0, domain: site.slug } );
+					fetchActions = createFetchActionsForModule( module, activeFilter, endDate, { post: queryOptions.post } );
```

Comment:
Check to see if this `post` param applies to both actions... I'm guessing not.

TODO: don't accept the extra params to `createFetchActionsForModule` & just explicitly modify the fetch actions here for exceptions.